src/layouts/views/WebcamView/WebcamView.js

```diff
@@ -129,15 +145,63 @@ class WebcamView extends React.Component {
     } else {
       context.drawImage(this.webcam, -115, -20, 1355, 756);
     }
-    const imageSrc = this.canvas.toDataURL('image/jpeg', 1.0);
-    addScan(component, imageSrc, currentStep, true);
-    this.setState({ saveImage: true });
+
+    this.stopRecording();
+
+    const blobCallback = (blob) => {
+      addScan(component, blob, currentStep, true);
+      this.setState({ saveImage: true });
+    };
+    this.canvas.toBlob(blobCallback, 'image/jpeg', 1.0);
+  }
+
+  stopRecording() {
+    this.setState({ recording: false });
+    if (this.mediaRecorders.length > 0) {
+      this.mediaRecorders.forEach((recorder) => { recorder.stop(); });
+    }
+  }
+
+  initVideoRecorder(stream, videoDuration) {
+    if (!this.state.recording) return;
+    const { addScan, currentStep } = this.props;
+    const startTime = Date.now() / 1000;
+    const mediaRecorder = new MediaRecorder(stream, { mimeType: 'video/webm\;codecs=vp8' });
+    this.mediaRecorders.push(mediaRecorder);
+
+    mediaRecorder.onstop = () => {
+      this.initVideoRecorder(stream, videoDuration);
+    };
+
+    mediaRecorder.ondataavailable = ({ data }) => {
+      // store chunk with data
+      const duration = Date.now() / 1000 - startTime;
```

Comment:
Move date.now / 1000 to state may be? Since it’s reused.

Reply:
This duration is video length in seconds, it is generating every time when video is captured and is saved together with video blob in `state.videoChunks`. We need this to be able to filter videos and get the most accepted one

Reply:
+1

Reply:
oh, I see, ny bad